cxx-qt-gen/src/syntax/attribute.rs

```diff
@@ -0,0 +1,144 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::syntax::path::path_compare_str;
+use std::iter::FromIterator;
+use syn::{
+    ext::IdentExt,
+    parenthesized,
+    parse::{Parse, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    token::Comma,
+    token::Paren,
+    Attribute, Error, Ident, Result,
+};
+
+/// Representation of a list of idents in an attribute, eg attribute(A, B, C)
+pub struct AttributeList {
+    pub paren: Paren,
+    pub items: Punctuated<Ident, Comma>,
+}
+
+impl Parse for AttributeList {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let content;
+        let paren = parenthesized!(content in input);
+        let items = content.parse_terminated(Ident::parse_any)?;
+        Ok(AttributeList { paren, items })
+    }
+}
+
+/// Returns the first attribute that matches a given path
+pub fn attribute_find_path<'a>(attrs: &'a [Attribute], path: &[&str]) -> Option<&'a Attribute> {
```

Comment:
No need to deal with this at the moment, but these seem like generally useful functions that might be nice to upstream into syn.

Reply:
So although these methods look useful to upstream syn, there are so many combinations of different nested items in syn types that it probably wouldn't be accepted. These methods i have here are helpers to reduce repeating code that CXX-Qt uses. And we basically make a load of assumptions to simplify the checks about what are inside the types.